### PR TITLE
Adding small test update for temp dir using t.TempDir

### PR DESCRIPTION
**Pull request by zedGGs**
This is a little fix update for this issue: 
https://github.com/prometheus/prometheus/issues/9594

Use case. Why is this important? 
Error handling is implicit.
Cleanup is implicit.

### CODE REVIEW

**Comment by zedGGs** [comment]
Im also pinging @beorn7 , for review if possible

File: promql/engine_test.go [hunk 1]
```diff
@@ -47,9 +47,10 @@ func TestMain(m *testing.M) {
 func TestQueryConcurrency(t *testing.T) {
 	maxConcurrency := 10
 
-	dir, err := os.MkdirTemp("", "test_concurrency")
-	require.NoError(t, err)
-	defer os.RemoveAll(dir)
+	dir := t.TempDir()
+	t.Cleanup(func() {
```
Comment (roidelapluie) [inline promql/engine_test.go:51]: t.TempDir() TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete. 
```suggestion
```

Comment (zedGGs) [reply promql/engine_test.go:51]: @roidelapluie thank you for the explanation, all make sense, I will update pr

**Review by jesusvazquez — APPROVED** [review]
LGTM Thank you for your contribution

File: tsdb/tsdbutil/dir_locker_testutil.go [hunk 2]
```diff
@@ -60,12 +60,8 @@ func TestDirLockerUsage(t *testing.T, open func(t *testing.T, data string, creat
...
 	for _, c := range cases {
 		t.Run(fmt.Sprintf("%+v", c), func(t *testing.T) {
-			tmpdir, err := os.MkdirTemp("", "test")
-			require.NoError(t, err)
-			t.Cleanup(func() {
-				require.NoError(t, os.RemoveAll(tmpdir))
-			})
-
+			tmpdir := t.TempDir()
+
```
Comment (bboreham) [inline tsdb/tsdbutil/dir_locker_testutil.go:64]: FYI this is a line consisting only of tabs, which fails current linting rules.

**Comment by bboreham** [comment]
`TestQueryConcurrency` is now failing consistently on Windows, e.g. https://github.com/prometheus/prometheus/actions/runs/7364591505/job/20045110499?pr=13343

I think I will revert this change until we sort out what has gone wrong.

**Pull request merged**